src/coreclr/CMakeLists.txt

```diff
@@ -202,11 +202,12 @@ if(CLR_CMAKE_HOST_UNIX)
     add_subdirectory(debug/createdump)
   endif(CLR_CMAKE_HOST_OSX OR (CLR_CMAKE_HOST_LINUX AND NOT CLR_CMAKE_HOST_UNIX_X86 AND NOT CLR_CMAKE_HOST_ANDROID))
 
-  # Include the dummy c++ include files
-  include_directories("pal/inc/rt/cpp")
-
-  # This prevents inclusion of standard C compiler headers
-  add_compile_options(-nostdinc)
+  # The CoreCLR PAL used to redefine NULL, which caused a number of null conversion and arithmetic
+  # warnings and errors to be suppressed.
+  # Suppress these warnings here to avoid breaking the build.
+  add_compile_options($<$<COMPILE_LANG_AND_ID:CXX,Clang,AppleClang>:-Wno-null-arithmetic>)
```

Comment:
Why do we have AppleClang as a separate compiler id?

Reply:
We need to specify it separately here as CMake's generator expressions don't allow us to do a regex match for the compiler ID; CMake only does exact matches here.

Reply:
I have actually meant something else - why do we have AppleClang here at all - this seems to be the only usage of AppleClang in the whole build system.

Reply:
In the rest of the build system, we use `CMAKE_<LANG>_COMPILER_ID MATCHES "Clang"` to match both Clang and AppleClang. As of CMake 3.0 (and CMake policy [CMP0025](https://cmake.org/cmake/help/latest/policy/CMP0025.html)), Apple's Clang build is a different ID. When we updated to a minimum CMake >= 3.0, we were able to use the `MATCHES` model for all cases where we were checking for Clang to easily support AppleClang.